File: modules/core/app/controllers/core/BlanksTestController.java
package controllers.core;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import model.blanks.BlanksExercise;
import model.exercise.Success;
import play.data.DynamicForm;
import play.data.FormFactory;
import play.libs.Json;
import play.mvc.Result;

public class BlanksTestController extends AbstractController {
  
  @Inject
  public BlanksTestController(FormFactory theFactory) {
    super(theFactory, "blanks");
  }
  
  public Result correctBlanks(int id) {
    DynamicForm form = factory.form().bindFromRequest();
    int inputCount = Integer.parseInt(form.get("count"));
    
    List<String> inputs = new ArrayList<>(inputCount);
    for(int count = 0; count < inputCount; count++)
      inputs.add(form.get("inp" + count));
    
    BlanksExercise exercise = new BlanksExercise(id);
    List<Success> results = exercise.correct(inputs);
    
    return ok(Json.toJson(results));
  }
  
  public Result testBlanks() {
    return ok(views.html.blanks.render(getUser(), new BlanksExercise(1)));
  }
}

File: app/controllers/blanks/BlanksTestController.scala
package controllers.blanks

import javax.inject._

import controllers.core.AIdExController
import model.blanks.{BlanksExercise, BlanksExerciseReads}
import model.core.result.{CompleteResult, EvaluationResult}
import model.core.{Repository, Secured}
import play.api.data.Form
import play.api.db.slick.DatabaseConfigProvider
import play.api.libs.json.Reads
import play.api.mvc.{ControllerComponents, EssentialAction}

import scala.concurrent.ExecutionContext
import scala.util.Try

class BlanksTestController @Inject()(cc: ControllerComponents, dbcp: DatabaseConfigProvider, r: Repository)(implicit ec: ExecutionContext)
  extends AIdExController[BlanksExercise, EvaluationResult](cc, dbcp, r, BlanksToolObject) with Secured {

  override def solForm: (Form[SolType]) = ???

  implicit def reads: Reads[BlanksExercise] = BlanksExerciseReads.blanksExReads

  override type TQ = repo.BlanksExercisesTable

  override def tq = repo.blanksExercises

  private val exercise = null // new BlanksExercise


  def correctBlanks(id: Int): EssentialAction = withUser { user =>
    implicit request =>
      //    val form = factory.form().bindFromRequest()
      //    val inputCount = Integer.parseInt(form.get("count"))
      //
      //    val inputs = (for (count <- 0 until inputCount) yield form.get(s"inp$count")).toList
      //
      //    ok(Json.toJson(exercise.correct(inputs)))
      Ok("TODO")
  }

  override protected def correctEx(sol: SolType, exercise: Option[BlanksExercise], user: model.User): Try[CompleteResult[EvaluationResult]] = ???

  def testBlanks: EssentialAction = withUser { user => implicit request => Ok(views.html.blanks.blanks.render(user, exercise)) }
}

